### PR TITLE
Argoproj promotions/membership

### LINKED ISSUE #276: REQUEST: Promotion to Approver for Joibel
### GitHub Username

@Joibel

### Sub-project(s)

Promotion is requested for the following sub-project(s). 

- [ ] Argo CD
- [ ] Argo Events
- [ ] Argo Rollouts
- [x] Argo Workflows

### Requirements

- [x] Reviewer for at least 3 months (as of the membership meeting date)
- [x] Reviewer for or author of at least 10 substantial PRs to the codebase
   - [Completed workflow tracking](https://github.com/argoproj/argo-workflows/pull/12198) which was a lot of pain to track down where workflow could run onExit despite having completed already.
   - Improving test coverage and suite: [test](https://github.com/argoproj/argo-workflows/pull/12216) [test](https://github.com/argoproj/argo-workflows/pull/12215) and [test](https://github.com/argoproj/argo-workflows/pull/12214) improvements for regression testing
   - A review [preventing](https://github.com/argoproj/argo-workflows/pull/12595#issuecomment-1918887327) a possibly misleading approach to retries in conflict
   - A review [which helped steer](https://github.com/argoproj/argo-workflows/pull/12574#issuecomment-1914255041) a fix into a long term solution for regressions in 3.5.3
   - Tackling use of [withItems with hooks](https://github.com/argoproj/argo-workflows/pull/12281) including regression test.
   - Investigation into [regression with reapplying workflow](https://github.com/argoproj/argo-workflows/issues/12601), including resolution.
   - Review of [memoization under retry](https://github.com/argoproj/argo-workflows/pull/12491) including local testing and catching regressions.
   - Helping eduardobr (#272) as reviewer of [stop strategy for CronWorkflows](https://github.com/argoproj/argo-workflows/pull/12305)
   - Helping shuankun (#266) get onboard as one of his earlier reviewers for [deleting pending pods on workflow termination](https://github.com/argoproj/argo-workflows/pull/12196)
  - Took devcontainer development forward so that it was generally usable, and have reviewed several further improvements to this 
    - https://github.com/argoproj/argo-workflows/pull/11104
    - https://github.com/argoproj/argo-workflows/pull/11115
    - As reviewer https://github.com/argoproj/argo-workflows/pull/11350
    - https://github.com/argoproj/argo-workflows/pull/11153
    - https://github.com/argoproj/argo-workflows/pull/11162
    - As reviewer https://github.com/argoproj/argo-workflows/pull/11981
  - Improved memoization so that it works in more cases, and based on popularity of the idea, changed it to allow 'no output' memoization.
    - https://github.com/argoproj/argo-workflows/pull/11225
    - https://github.com/argoproj/argo-workflows/pull/11257
    - https://github.com/argoproj/argo-workflows/pull/11379
    - https://github.com/argoproj/argo-workflows/pull/12119
  - Made retryPolicy saner so that it behaves more like people would expect. Included diagramming in the documentation. https://github.com/argoproj/argo-workflows/pull/11005
  - Improved how hook expressions are handled https://github.com/argoproj/argo-workflows/pull/10919
  - https://github.com/argoproj/argo-workflows/pull/11151
  - Allowed hooks to be added globally in workflowDefaults https://github.com/argoproj/argo-workflows/pull/11214 and then fixed it in https://github.com/argoproj/argo-workflows/pull/11693

Other reviews where I've contributed to changes across the controller codebase:
  - https://github.com/argoproj/argo-workflows/pull/11736
  - https://github.com/argoproj/argo-workflows/pull/11760
  - https://github.com/argoproj/argo-workflows/pull/11456
  - https://github.com/argoproj/argo-workflows/pull/11350
  - https://github.com/argoproj/argo-workflows/pull/10777

Author of [35](https://github.com/argoproj/argo-workflows/pulls?q=is%3Apr+author%3AJoibel+is%3Aclosed) merged PRs, 11 since becoming a reviewer.

**Other notes**

I've been trying to build the community around workflows up in line with @juliev0 sustainability efforts, so my time has been spent more in slack and issues than PRs recently. I've helped with getting new people interested in membership of argoproj.

In the last [3 months](https://argo.devstats.cncf.io/d/48/users-statistics-by-repository-group?orgId=1&var-period=w&var-metric=contributions&var-repogroup_name=argoproj%2Fargo-workflows&var-users=All&from=now-3M&to=now) since becoming a reviewer I've been 4th most active in the github project.

image

I am very active in #argo-workflows (and quite active in #argo-cd) slack channels.

### Lead Sponsor(s)

Promotion to Approver requires sponsorship from at least one sub-project Lead.

- @terrytangyuan 

### Other Sponsors

- @juliev0

### LINKED ISSUE #283: REQUEST: Promotion to Reviewer for @agaudreault
### GitHub Username

@agaudreault-jive

### Sub-project(s)

Promotion is requested for the following sub-project(s).

- [x] Argo CD
- [ ] Argo Events
- [ ] Argo Rollouts
- [ ] Argo Workflows

### Requirements

- [x] Member for at least 3 months (as of the membership meeting date)
- [x] Reviewer for or author of at least 5 substantial PRs to the codebase
    - (Author) https://github.com/argoproj/argo-cd/pull/13912
    - (Author) https://github.com/argoproj/argo-cd/pull/16820
    - (Author) https://github.com/argoproj/argo-cd/pull/16781
    - (Review/Author) https://github.com/argoproj/argo-cd/pull/15237
    - (Review) https://github.com/argoproj/argo-cd/pull/15636
    - (Review) https://github.com/argoproj/argo-cd/pull/15137
    - (Review) https://github.com/argoproj/argo-cd/pull/15741
    - (Review) https://github.com/argoproj/argo-cd/pull/17650
    - (Review) https://github.com/argoproj/argo-cd/pull/17275
    - (Review) https://github.com/argoproj/argo-cd/pull/17808
    - All PRs/Reviews:
      - [Contributions](https://github.com/argoproj/argo-cd/pulls?q=is%3Apr+author%3Aagaudreault+is%3Amerged)
      - [Reviews](https://github.com/argoproj/argo-cd/pulls?q=is%3Apr+reviewed-by%3Aagaudreault+-author%3Aagaudreault+is%3Aclosed+)

### Lead Sponsor(s)

Promotion to Reviewer requires sponsorship from at least one sub-project Lead.

-

### Other Sponsors

- @crenshaw-dev 
- @leoluz

### LINKED ISSUE #291: REQUEST: New membership for OpenGuidou
### GitHub Username

@OpenGuidou

### Requirements
- [x] I have reviewed [the community membership guidelines](https://github.com/argoproj/argoproj/blob/master/community/membership.md)
- [x] I have [enabled 2FA on my GitHub account](https://github.com/settings/security)
- [x] I am actively contributing to 1 or more Argoproj subprojects
- [x] I have two sponsors that meet the sponsor requirements listed in the community membership guidelines
- [x] I have spoken to my sponsors ahead of this application, and they have agreed to sponsor my application
- [x] I understand that I can [make my membership public](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-membership-in-organizations/publicizing-or-hiding-organization-membership) if I'd like to once I am invited to the organization

### Sponsors
- @zachaller 
- @crenshaw-dev 

### List of contributions to the Argoproj project
- PRs reviewed / authored:
https://github.com/argoproj/argo-rollouts/pull/3353
https://github.com/argoproj/argo-rollouts/pull/3213
https://github.com/argoproj/argo-rollouts/pull/3038
https://github.com/argoproj/argo-rollouts/pull/2914
- Issues responded to
- Projects I am involved with: Argo CD, Argo Rollouts

### LINKED ISSUE #293: REQUEST: Promotion to Reviewer for @shuangkun
### GitHub Username

@shuangkun 

### Sub-project(s)

Promotion is requested for the following sub-project(s).

- [ ] Argo CD
- [ ] Argo Events
- [ ] Argo Rollouts
- [x] Argo Workflows

### Requirements

- [x] Member for at least 3 months (as of the membership meeting date)
- [x] Reviewer for or author of at least 5 substantial PRs to the codebase
Authored：
    - < offload the envtemplate to support long arguments> https://github.com/argoproj/argo-workflows/pull/12325
    - < parallel resolve reference to speed up execution of large workflow > https://github.com/argoproj/argo-workflows/pull/12328
    - < update command for cron/wft/cwft > https://github.com/argoproj/argo-workflows/pull/12803
    - < move delete logic from server to controller > https://github.com/argoproj/argo-workflows/pull/12734
    - < enable scheduling constraints in step/dag level > https://github.com/argoproj/argo-workflows/pull/12700
    -  < enable dynamic template reference naming > https://github.com/argoproj/argo-workflows/pull/12842
    - < raise and resolve issues in large workflow > 
              https://github.com/argoproj/argo-workflows/pull/12574
              https://github.com/argoproj/argo-workflows/pull/12537
              https://github.com/argoproj/argo-workflows/pull/11048
    - < locate the causes of some very important stuck issues about workflow/containerset/terminating >
               https://github.com/argoproj/argo-workflows/pull/12196
               https://github.com/argoproj/argo-workflows/pull/12761
               https://github.com/argoproj/argo-workflows/pull/12756
     - < fixed some issues with core workflow logict about retryStrategy/memoize/hook/loop/dag/containerOn/artifacts/http/exector/data/resouce/server/controller>
                https://github.com/argoproj/argo-workflows/pull/12491
                https://github.com/argoproj/argo-workflows/pull/12683
                https://github.com/argoproj/argo-workflows/pull/12815
 	        https://github.com/argoproj/argo-workflows/pull/12701
                https://github.com/argoproj/argo-workflows/pull/12817
                https://github.com/argoproj/argo-workflows/pull/12835
                https://github.com/argoproj/argo-workflows/pull/12831
                https://github.com/argoproj/argo-workflows/pull/12825
                https://github.com/argoproj/argo-workflows/pull/12780
                https://github.com/argoproj/argo-workflows/pull/12575
                https://github.com/argoproj/argo-workflows/pull/12544
                https://github.com/argoproj/argo-workflows/pull/12419
                https://github.com/argoproj/argo-workflows/pull/12197
                https://github.com/argoproj/argo-workflows/pull/12084

[Totol Authored 52](https://github.com/argoproj/argo-workflows/pulls/shuangkun) / [44 Merged](https://github.com/argoproj/argo-workflows/pulls?q=is%3Apr+author%3Ashuangkun+is%3Aclosed) 

[Reviewed](https://github.com/argoproj/argo-workflows/pulls?page=1&q=is%3Apr+is%3Aclosed+commenter%3Ashuangkun)

[Issues](https://github.com/argoproj/argo-workflows/issues/assigned/shuangkun)

Other Notes:

 I am the fourth in the past [6months](https://argo.devstats.cncf.io/d/48/users-statistics-by-repository-group?orgId=1&var-period=w&var-metric=contributions&var-repogroup_name=argoproj%2Fargo-workflows&var-users=All&from=now-6M&to=now) on the CNCF DevStats for Argo Workflows.
image
I have over three years of experience using and troubleshooting Argo Workflows issues on my team. My team also participated in the Argo workflow community contribution very early and member has participated in maintaining it before. Hoping to make more contributions in the future.

### Lead Sponsor(s)

Promotion to Reviewer requires sponsorship from at least one sub-project Lead.

- @ terrytangyuan 

### Other Sponsors

- @ tczhao 
- @ juliev0

### LINKED ISSUE #294: REQUEST: New membership for jiachengxu
### GitHub Username

@jiachengxu

### Requirements
- [x] I have reviewed [the community membership guidelines](https://github.com/argoproj/argoproj/blob/master/community/membership.md)
- [x] I have [enabled 2FA on my GitHub account](https://github.com/settings/security)
- [x] I am actively contributing to 1 or more Argoproj subprojects
- [x] I have two sponsors that meet the sponsor requirements listed in the community membership guidelines
- [x] I have spoken to my sponsors ahead of this application, and they have agreed to sponsor my application
- [x] I understand that I can [make my membership public](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-membership-in-organizations/publicizing-or-hiding-organization-membership) if I'd like to once I am invited to the organization

### Sponsors
- @agilgur5 
- @jessesuen 
- @terrytangyuan 

### List of contributions to the Argoproj project
- PRs reviewed / authored
- argo-workflows:
  - https://github.com/argoproj/argo-workflows/pull/12912
  - https://github.com/argoproj/argo-workflows/pull/12736
  - https://github.com/argoproj/argo-workflows/pull/10693
  - https://github.com/argoproj/argo-workflows/pull/10677
  - https://github.com/argoproj/argo-workflows/pull/10489
  - https://github.com/argoproj/argo-workflows/pull/10449
  - https://github.com/argoproj/argo-workflows/pull/10424
  - https://github.com/argoproj/argo-workflows/pull/10346  
- argo-cd:
  - https://github.com/argoproj/argo-cd/pull/13360
- argo-rollouts:
  - https://github.com/argoproj/argo-rollouts/pull/2557
- Issues responded to
- Projects I am involved with

### LINKED ISSUE #295: REQUEST: New membership for @jswxstw
### jswxstw

@jswxstw

### Requirements
- [x] I have reviewed [the community membership guidelines](https://github.com/argoproj/argoproj/blob/master/community/membership.md)
- [x] I have [enabled 2FA on my GitHub account](https://github.com/settings/security)
- [x] I am actively contributing to 1 or more Argoproj subprojects
- [x] I have two sponsors that meet the sponsor requirements listed in the community membership guidelines
- [x] I have spoken to my sponsors ahead of this application, and they have agreed to sponsor my application
- [x] I understand that I can [make my membership public](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-membership-in-organizations/publicizing-or-hiding-organization-membership) if I'd like to once I am invited to the organization

### Sponsors
- @agilgur5 
- @Joibel 

### List of contributions to the Argoproj project
- PRs reviewed / authored
  - https://github.com/argoproj/argo-workflows/pull/12440
  - https://github.com/argoproj/argo-workflows/pull/12620
  - [Other open PRs](https://github.com/argoproj/argo-workflows/pulls/jswxstw)
- Issues responded to
  - https://github.com/argoproj/argo-workflows/issues/12733
  - https://github.com/argoproj/argo-workflows/issues/12760
  - https://github.com/argoproj/argo-workflows/issues/12765
  - https://github.com/argoproj/argo-workflows/issues/12779
  - https://github.com/argoproj/argo-workflows/issues/12790
  - https://github.com/argoproj/argo-workflows/issues/12821
  - [Full list](https://github.com/argoproj/argo-workflows/issues?q=involves%3Ajswxstw+is%3Aissue)
- Projects I am involved with
  - [Argo Workflows](https://github.com/argoproj/argo-workflows)

### LINKED ISSUE #296: REQUEST: Promotion to Approver for @tczhao
### GitHub Username

@tczhao 

### Sub-project(s)

Promotion is requested for the following sub-project(s). 

- [ ] Argo CD
- [ ] Argo Events
- [ ] Argo Rollouts
- [x] Argo Workflows

### Requirements

- [x] Reviewer for at least 3 months (as of the membership meeting date)
- [x] Reviewer for or author of at least 10 substantial PRs to the codebase
   - https://github.com/argoproj/argo-workflows/pull/12734
   - https://github.com/argoproj/argo-workflows/pull/12842
   - https://github.com/argoproj/argo-workflows/pull/12756
   - https://github.com/argoproj/argo-workflows/pull/12815
   - https://github.com/argoproj/argo-workflows/pull/12701
   - https://github.com/argoproj/argo-workflows/pull/12817
   - https://github.com/argoproj/argo-workflows/pull/12792
   - https://github.com/argoproj/argo-workflows/pull/12799
   - https://github.com/argoproj/argo-workflows/pull/8993
   - https://github.com/argoproj/argo-workflows/pull/9538
   - https://github.com/argoproj/argo-workflows/pull/8779

[~90 PRs](https://github.com/argoproj/argo-workflows/pulls?q=author%3Atczhao) opened
[~90 PRs](https://github.com/argoproj/argo-workflows/pulls?q=is%3Apr+commenter%3Atczhao+-author%3Atczhao+) responded to or reviewed
[3rd in activity from past 2 months](https://argo.devstats.cncf.io/d/48/users-statistics-by-repository-group?orgId=1&var-period=w&var-metric=contributions&var-repogroup_name=argoproj%2Fargo-workflows&var-users=All&from=now-2M&to=now)

Been contributing on and off since 2021, and have a reasonable sense of areas/practices to look out for in reviewing PRs

### Lead Sponsor(s)

Promotion to Approver requires sponsorship from at least one sub-project Lead.

- @terrytangyuan 

### Other Sponsors

- @juliev0
- @agilgur5 
- @shuangkun

### PR DESCRIPTION
**Pull request by crenshaw-dev**
# Argoproj promotions/membership

Every quarter, the Argoproj maintainers review requests for project membership and promotions (read about the roles [here](https://github.com/argoproj/argoproj/blob/master/community/membership.md#community-membership)). The changes below recognize the incredible efforts and commitment to the project by these people. Thank you so much!

## 🚀  Maintainer promotions

Congrats to the following contributors for their promotions:

* Alan Clucas (@Joibel) to Approver in Argo Workflows
* Tianchu Zhao (@tczhao) to Approver in Argo Workflows
* Shuangkun Tian (@shuangkun) to Reviewer in Argo Workflows
* Alexandre Gaudreault (@agaudreault) to Reviewer in Argo CD

## ⭐  New members

An additional congratulations to the following folks for becoming Argoproj members:

* Jiacheng Xu	(@jiachengxu)
* @jswxstw
* @OpenGuidou

You all will receive invitations on GitHub to become members of the Argoproj organization.

## 🎓 Alumni

The maintainers document contains a new section for folks who were previously maintainers but have been inactive for at least a year. While not currently active, they deserve continued recognition for their impact on the project!

Closes #276
Closes #283
Closes #291
Closes #293 
Closes #294 
Closes #295 
Closes #296

### CODE REVIEW

**Comment by terrytangyuan** [comment]
I'll grant write access to new approvers for Argo Workflows repo.

Congratulations everyone!

**Review by terrytangyuan — APPROVED** [review]
🎉

**Review by terrytangyuan — COMMENTED** [review]
@crenshaw-dev Would you like to update https://github.com/argoproj/argoproj/blob/main/OWNERS too?

File: MAINTAINERS.md [hunk 1]
```diff
@@ -52,19 +52,21 @@
...
 | Hong Wang                 | [wanghong230](https://github.com/wanghong230)           | Reviewer                                               | [Akuity](https://akuity.io/)                         |
 | Jonathan West             | [jgwest](https://github.com/jgwest)                     | Approver - CD                                          | [Red Hat](https://www.github.com/redhat/)            |
 | May Zhang                 | [mayzhang2000](https://github.com/mayzhang2000)         | Approver - CD                                          | [Intuit](https://www.github.com/intuit/)             |
-| Tianchu Zhao              | [tczhao](https://github.com/tczhao)                     | Reviewer - Workflows                                   | Independent                                          |
+| Tianchu Zhao              | [tczhao](https://github.com/tczhao)                     | Approver - Workflows                                   | Independent                                          |
```
Comment (tczhao) [inline MAINTAINERS.md:61]: Hi @crenshaw-dev ,
Thank you for organising this,
Could you update this line to
```
| Tianchu Zhao | [tczhao](https://github.com/tczhao) | Approver - Workflows | [Atlan](https://atlan.com/) |
```

**Review by tczhao — CHANGES_REQUESTED** [review]

File: MAINTAINERS.md (hunk 1 above)
Comment (crenshaw-dev) [reply MAINTAINERS.md]: Done, and assuming it's a new role, congrats! :-)

**Comment by crenshaw-dev** [comment]
Looks like all the related tasks are done. Please let me know if we missed anything!

**Pull request merged**